errbot/backends/test.py

```diff
@@ -461,6 +461,26 @@ def assertCommandFound(self, command, timeout=5):
         self.bot.push_message(command)
         assert 'not found' not in self.bot.pop_message(timeout)
 
+    def inject_mocks(self, PluginName: str, MockDict: dict):
+        """Inject mock objects into the plugin
+
+        mock_dict = {
+            'field_1': obj_1,
+            'field_2': obj_2,
+        }
+        testbot.inject_mocks(hello, mock_dict)
+        assert 'blah' in exec_command('!hello')
+        """
+        plugin = self.bot.plugin_manager.get_plugin_obj_by_name(PluginName)
+
+        if plugin is None:
+            raise Exception(f'"{PluginName}" is not loaded.')
+        for field, mock_obj in MockDict.items():
+            if hasattr(plugin, field):
```

Comment:
fail early:
```python
if not hasattr(plugin, field):
     raise ValueError(f'No property/attribute named "{field}" attached.')

setattr(plugin, field, mock_obj)
```

Reply:
@gbin the testbot is not able to load the DummyPlugin and that is the reason tests are failing. Am I missing something in the logic here? I tried both ways to fetch the loaded plugin:
```
self.bot.plugin_manager.get_plugin_obj_by_name(PluginName)
```
as well as:
```
self.bot.plugin_manager.plugins[PluginName]
```